endpoint.php

```diff
@@ -254,14 +254,46 @@ function invalidRequest(): void
         exit();
     }
     $revoke = filter_input(INPUT_POST, 'action', FILTER_VALIDATE_REGEXP, ['options' => ['regexp' => '@^revoke$@']]);
+    $token = filter_input(INPUT_POST, 'token', FILTER_VALIDATE_REGEXP, ['options' => ['regexp' => '@^[0-9a-f]+_[0-9a-f]+$@']]);
+    // check if is POST+revoke request
     if (is_string($revoke)) {
-        $token = filter_input(INPUT_POST, 'token', FILTER_VALIDATE_REGEXP, ['options' => ['regexp' => '@^[0-9a-f]+_[0-9a-f]+$@']]);
         if (is_string($token)) {
             revokeToken($token);
         }
         header('HTTP/1.1 200 OK');
         exit();
     }
+    // check if is POST+introspection request
+    if (is_string($token)) {
+        // TODO: authorize resource server as per specification (see https://indieauth.spec.indieweb.org/#access-token-verification-response-p-1)
+        // meaning verify that the Basic user is the client_id, and ignore the Basic password
+        $tokenInfo = retrieveToken($token);
+        if (!isset($tokenInfo)) {
+            exit(json_encode([
+                'token_type' => 'Bearer',
+                'me' => '',
+                'sub' => '',
+                'client_id' => '',
+                'aud' => '',
+                'scope' => '',
+                'iat' => time(),
+                'exp' => time(),
+                'active' => false,
+            ]));
```

Comment:
When `active` is `false`, it should not return any other JSON members. Per [Example 15 in the current IndieAuth spec](https://indieauth.spec.indieweb.org/20240711/#example-15) and [section 2.2. Introspection Response](https://www.rfc-editor.org/rfc/rfc7662#section-2.2) from RFC 7662.

```suggestion
            exit(json_encode([
                'active' => false,
            ]));
```

Reply:
Totally missed that in the specifications! Will do.